test/unit/org/apache/cassandra/cql3/validation/entities/TimestampTest.java

```diff
@@ -97,6 +97,42 @@ public void testTimestampTTL() throws Throwable
                    row(1, null, null));
     }
 
+    @Test
+    public void testMaxTimestamp() throws Throwable
+    {
+        String myType = createType("CREATE TYPE %s (a int, b int)");
+        createTable("CREATE TABLE %s (k int PRIMARY KEY, a text, " +
+                    "l list<int>, fl frozen<list<int>>," +
+                    "s set<int>, fs frozen<set<int>>," +
+                    "m map<int, text>, fm frozen<map<int, text>>," +
+                    "t " + myType + ", ft frozen<" + myType + ">)");
+        execute("INSERT INTO %s (k, a, l, fl, s, fs, m, fm, t, ft) VALUES " +
+                "(1, 'test', [1], [2], {1}, {2}, {1 : 'a'}, {2 : 'b'}, {a : 1, b : 1 }, {a : 2, b : 2}) USING TIMESTAMP 1");
+        // update the list, so its maxWritetime should be more advanced than other columns
+        execute("UPDATE %s USING TIMESTAMP 2 SET l = l + [10] WHERE k = 1");
```

Comment:
This only exercises the selection among multiple values for the list. It could be useful to test all the multicell types, probably adding a separate update and timestamp per type.